pkgs/applications/editors/android-studio/common.nix

```diff
@@ -80,6 +81,7 @@ let
         --set-default JAVA_HOME "$out/jre" \
         --set ANDROID_EMULATOR_USE_SYSTEM_LIBS 1 \
         --set QT_XKB_CONFIG_ROOT "${xkeyboard_config}/share/X11/xkb" \
+        ${if tiling_wm then "--set _JAVA_AWT_WM_NONREPARENTING 1"  else ""} \
```

Comment:
My main concern is if this is the right way to do this. Seems a little messy

Reply:
I have not done this conditionals but another way would be to add it at the end as an `optional`.

Examples from nixpkgs:

```
57-  ''
58:   + (lib.optionalString (lisp-compiler.pname == "ecl") ''
59-     cp src/binary-ecl/maxima.fas* "$out/lib/maxima/${version}/binary-ecl/"
60-   '')
```

```
64:  ) ++ lib.optional enableCuda "-DGMX_GPU=CUDA";
```

```
39-    "--with-gmp=${lib.getDev gmp}"
40-    "--with-readline=${lib.getDev readline}"
41-  ]
42:  ++ lib.optional stdenv.isDarwin "--host=x86_64-darwin"
43:  ++ lib.optional withThread "--mt=pthread";
```

Though, not really  sure about best practices so can't help much more than pointing some of this examples.

Reply:
ah yeah, that is a little nicer.